### PR TITLE
[PVR] Fix PVR windows handling for action "back".

### PR DESCRIPTION
**Pull request by ksooo**
As discussed on Slack, special handling of PVR windows for action "back" is buggy and not in sync with the window handling (history) of the rest of Kodi. This PR aligns PVR handling for action "back" with the UX of the rest of the Kodi windows.

One of the bugs fixed:
* Go to LiveTV in the main menu (Estuary Home screen)
* This brings up the guide window
* Press "left" to get to the side menu
* Chose the "channels" icon
* This brings me to the channels window
* Now press back => you end in the main menu (Estuary Home screen), not in the guide window

Runtime-tested on macOS and Android, latest Kodi master.

@a1rwulf, @phunkyfish  for code review?

### CODE REVIEW

**Review by a1rwulf — APPROVED** [review]
LGTM.
Runtime tested on Linux.

File: xbmc/pvr/windows/GUIWindowPVRChannels.cpp [hunk 1]
```diff
@@ -57,6 +57,16 @@ CGUIWindowPVRChannelsBase::~CGUIWindowPVRChannelsBase()
       this);
 }
 
+std::string CGUIWindowPVRChannelsBase::GetRootPath() const
+{
+  //! @todo make CGUIMediaWindow::GetRootPath() non-const.
```
Comment (enen92) [inline xbmc/pvr/windows/GUIWindowPVRChannels.cpp:62]: can you explain why?

Comment (ksooo) [reply xbmc/pvr/windows/GUIWindowPVRChannels.cpp:62]: Sure. we are facing the old problem that somebody invents an interface - a virtual function - and makes an assumption about the implementation by defining the virtual function const. Now, the implementation has has either to provide the data from a member (eager evaluation enforced) or to do lazy evaluation, calculating the data on demand, but to be able to store the data in a member from within a const function, the member needs to be defined `mutable`. For this case here, I even need to call functions here doing the lazy data evaluation, which cannot be const by design. Thus, I have no other choice than to const_cast this.

There can be different opinions on this. One could say `GetRootPath` is **logically** const, thus the const_cast in the implementation is perfectly fine. I tend to say, the declaration of the virtual function needs to be revised for the reasons I gave above. There is no right and wrong, though. Mayby I should remove the comment.

Comment (enen92) [reply xbmc/pvr/windows/GUIWindowPVRChannels.cpp:62]: Thanks for explaining, I now understand the reason. Keeping the todo is fine to me, I'd rephrase it as the way it's written makes it sound like mandatory change and, as you say, there's not a right and wrong here (needs a careful evaluation).

Comment (ksooo) [reply xbmc/pvr/windows/GUIWindowPVRChannels.cpp:62]: Okay, I will rephrase the comment to be more a question.

**Review by enen92 — APPROVED** [review]

File: xbmc/pvr/windows/GUIWindowPVRChannels.cpp (hunk 1 above)
Comment (a1rwulf) [reply xbmc/pvr/windows/GUIWindowPVRChannels.cpp]: Yeah I guess there's no right and wrong, but I tend to be in the pro const side here.
For me the name `GetRootPath` makes it obvious it should be const precisely to not have side effects like https://github.com/xbmc/xbmc/blob/4e291ab51b6b646fcbc110890592271d60d8120f/xbmc/pvr/windows/GUIWindowPVRBase.cpp#L465 being called.

At this time, `GetRootPath` modifies `m_vecItems` path, which imho is a brutal side effect for a "getter".
